tests/test_spstats.py

```diff
@@ -171,3 +194,30 @@ def test_patches_method(self):
             gsd=diff.res[0],
             area_size=10000
         )
+
+class TestBinning:
+
+    def test_nd_binning(self):
+
+        ref, diff, mask = load_ref_and_diff()
+
+        slp, asp = xdem.coreg.calculate_slope_and_aspect(ref.data.squeeze())
+
+        # 1d binning
+        df = xdem.spstats.nd_binning_scipy(dh=diff.data.flatten(),list_var=[slp.flatten()],list_var_names=['slope'])
+
+        # nmad goes up quite a bit with slope, we can expect a 10 m measurement error difference
```

Comment:
I'm not really sure that is the kind of core functionality of the function you want to test.
What you should check is that the output has the right dimensions and range (let's say you have a binning of 10, the output should have a length 10, and the min/max values are known.

Reply:
Added